src/main.ts

```diff
@@ -657,12 +657,22 @@ export async function getToken(input: CspInput): Promise<string> {
 export async function loadAllData(executionGraph: Object): Promise<string[]> {
   let files: string[] = []
 
+  const onlyUploadOnFailure = core.getInput("only-upload-on-failed-tasks")
+
   //TODO assertions
   for (const task of executionGraph["tasks"]) {
     if (task["status"] === "SKIPPED") {
       continue
     }
 
+    if (task["status"] === "PASSED" && onlyUploadOnFailure === "true") {
+      continue
+    } else if (onlyUploadOnFailure === "false") {
+      core.debug(
```

Comment:
This message is good but it should be outside the loop. Otherwise is going to be redundantly printed a lot of times.